### DISCUSSION TITLE
Helicopter Control and Basic Understanding

### DISCUSSION THREAD

**Discussion by nmfarras** [JSBSim-Team/jsbsim#1066] [Q&A]
Hello, I have question again regarding Helicopter simulation using JSBSim. Right now I try AH1S and AH64D model using JSBSim in Unreal Engine environment. So far, suggestion from #659 help alot in my case. 

First, I already implement the attitude control as suggest by @gallonmate and its work as intended. The pitch, yaw, and roll command using stick still work, therefore i still use the example unreal project input for attitude command. However I still struggled in understanding how to use collective. Idk if I get it right, but i heard it is like cruise control in car where car can move forward and backward by changing gear and step on gas, but for collective case it goes up and down. Is my understanding so far is right?

Next question is, I try to implement like the updated readme for the collective input as following

![image](https://github.com/JSBSim-Team/jsbsim/assets/30061607/555ee8fb-b718-403f-8123-6321fef6d497)

The rpm input work i guess, but for collective i don't get any output. Is there any suggestion for this? Like do I need to use input with 0-1 axis or -1 to 1 axis value? Cause my helicopter keep flying with the rpm set like the screenshot below.

![image](https://github.com/JSBSim-Team/jsbsim/assets/30061607/bb933e24-2ae9-4871-95e2-e478bb5780a6)

One more thing, if you can kindly enough suggest what should be change when I want to accelerate or decelerate after pitching so i can move forward or backward will be much appreciated. Thank you!

Note: I might misunderstood rpm governor term cause I still get confused with the xml input since it is difference from typical one in aircraft.

**Comment by gallonmate** [comment]
The readme file was just an example how to use the blueprint nodes in general, it's not correct or perfect for the AH1S, but here's a better picture:  https://github.com/JSBSim-Team/jsbsim/discussions/861

For the A1HS:
The collective input should be 0-1. The collective only controls the amount of lift that's generated. 0 means no lift and 1 means full lift.
The other controls for the helicopter are taken from the regular aircraft inputs that control the pitch, roll, and yaw of the helicopter:
Elevator input tilts the rotor blade forward/backwards which makes the helicopter tilt forwards or backwards.
Aileron input tilts the rotor blade sideways which rolls the helicopter left or right.
Rudder input controls the tail rotor to spin the helicopter in place left or right.

I have no idea if the controls of the AH64D are mapped the same or how it works, as I don't see it in the JSBSim library. I only tested the AH1S.

**Reply by nmfarras** [reply]
Now I can pretty much fly it by adjusting throttle and rpm for the ah64d. I still don't get much different reaction when I set collective to 0 and when I put it 0.1 

And I also already follow how to setup the input like the issue #861. Maybe will try other model for comparison. 

I need to read more about helicopter flight dynamic too I guess. 

Anyway, thank you as always! Will update if I find anything.

**Comment by bomber64** [comment]
"Elevator input tilts the rotor blade forward/backwards which makes the helicopter tilt forwards or backwards.
Aileron input tilts the rotor blade sideways which rolls the helicopter left or right."

That's interesting.

Have a look at the UH-1 flight model here

https://github.com/FGMEMBERS-NONGPL/UH-1

It's not ya conventional flight model, but it is the best JSBsim helicopter flight model.

It's licenced CC non-commercial not GPL so might not suit you if you wish to sell it.

**Reply by nmfarras** [reply]
That's interesting. Thank you for the suggestion. Cause I found from this forum the Yasim ah1s seems to used some data from the NASA report about uh-1 https://forum.flightgear.org/viewtopic.php?t=13651#p139380

**Reply by bomber64** [reply]
If you take a look at their AH-1, you'll notice there's also a JSBsim Avatar Scorpion gunship

**Comment by 140695809** [comment]
I also want to try AH64D model using JSBSim in Unreal Engine environment. Can you give me some references?Thank you.
